core/shared/src/main/scala/zio/prelude/These.scala

```diff
@@ -1,77 +1,512 @@
 package zio.prelude
 
+import zio.NonEmptyChunk
 import zio.prelude.These._
 
+/**
+ * `These` is a data type representing a value that may either be a `Left`
+ * with an`A`, a `Right` with a `B`, or a `Both` with an `A` and a `B`.
+ *
+ * `These` can be useful to model certain domains where both values may be
+ * present in addition to one or the other. For example, in streaming
+ * applications we may want to consume values from two upstream producers
+ * concurrently. Depending on the timing of the producers either one producer,
+ * the other, or both producers may have values that are ready to be consumed.
+ * Using `These` provides a convenient way to model all of these possibilities
+ * in a single "flat" data type.
+ *
+ * `These` can also be useful for representing computations that may produce
+ * both a value and an error. For example, transferring money between two bank
+ * accounts might either succeed, fail completely if there are not sufficient
+ * funds, or succeed with some warnings if the sender's account balance would
+ * be very low after the transfer or the receiver's account has not been
+ * verified. `These` allows modeling these types of computations and preserving
+ * information regarding all errors while still potentially returning a
+ * successful computation.
+ */
 sealed trait These[+A, +B] { self =>
-  def combine[A1 >: A, B1 >: B](that: These[A1, B1])(implicit A1: Associative[A1], B1: Associative[B1]): These[A1, B1] =
```

Comment:
We might consider keeping the `combine` method, it would be useful for defining many type class instances.